### PR TITLE
tiny fixes and a cleanup

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by evverx** [comment]
This pull request **fixes 1 alert** when merging d61d832ac020a1007e3c02a30512050848c7cac6 into e094eb11888d984f1e9139675a5780a49c61d5ed - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-48c2ed13df52060a0d4d143f030b264ac8b174a8)

**fixed alerts:**

* 1 for Missing header guard

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by evverx** [comment]
I'm not sure why the alert popped up here because the issue it's about seems to have been introduced in a42d7cf16515ab0ab back in March and has nothing to do with this PR. @samlanning do you know why alerts like this (perfectly valid I must say) can appear in unrelated PRs?

File: src/boot/efi/util.h [hunk 1]
```diff
@@ -32,11 +32,12 @@ CHAR16 *stra_to_path(CHAR8 *stra);
...
-#endif
 
 static inline void FreePoolp(void *p) {
         FreePool(*(void**) p);
 }
 
 #define _cleanup_(x) __attribute__((cleanup(x)))
 #define _cleanup_freepool_ _cleanup_(FreePoolp)
+
+#endif
```
Comment (evverx) [inline src/boot/efi/util.h:43]: It's consistent with the rest of the headers in `src/boot/efi` and works totally fine, but it'd probably make sense to start using `#pragma once` sometime in the future.

**Comment by evverx** [comment]
@samlanning I'm sorry. I misinterpreted the comment my alter ego had posted :-) I though it was about introducing the issue, while, in fact, it was about fixing it.

File: src/core/dynamic-user.c [hunk 2]
```diff
@@ -34,7 +34,7 @@ static DynamicUser* dynamic_user_free(DynamicUser *d) {
 }
 
 static int dynamic_user_add(Manager *m, const char *name, int storage_socket[2], DynamicUser **ret) {
-        DynamicUser *d = NULL;
+        _cleanup_free_ DynamicUser *d = NULL;
```
Comment (evverx) [inline src/core/dynamic-user.c]: I think it's a bit dangerous to free `d` automatically especially after it has been put in the hash. Probably the current approach is safer in the long term. Should something go wrong, a memory leak is better than a double-free :-)

**Comment by samlanning** [comment]
@evverx No problem! That's an indicator that we should make our comments clearer though! :)

**Comment by evverx** [comment]
@samlanning I think comments with the word "fixes" in bold are pretty clear. It was just that I was jumping between two PRs in the morning and everything was blurry until I got a cup of coffee :-)

**Comment by yuwata** [comment]
@evverx I do not understand the possibility of double free in your comment about `DynamicUser`. But in the revised commit, I've dropped the `_cleanup_` attribute. PTAL.

**Comment by yuwata** [comment]
@evverx Thanks. I tagged this 'good to merge'.

**Comment by evverx** [comment]
This pull request **fixes 1 alert** when merging a2b635eb3940f33f5dd308d85ba416f0630d152b into 030836923d9f8600926b2a7dd24b249a540c31e6 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-7af54cf43c11bd986295395492bdde0f8867de28)

**fixed alerts:**

* 1 for Missing header guard

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/boot/efi/util.h (hunk 1 above)
Comment (keszybz) [reply src/boot/efi/util.h:43]: Yep, we should.

**Comment by evverx** [comment]
@samlanning it seems that LGTM restarts analysis every time labels are added or removed. Do you know if there is a way to turn it off?

**Comment by evverx** [comment]
I've just checked the "settings" tab and it turns out that GitHub doesn't allow me to send only "the code has been changed" events. Apparently, everything that happens within PRs is sent to LTGM.

**Comment by samlanning** [comment]
@evverx LGTM gets an event whenever there is a change to the PR from github. LGTM will only start a new analysis if the base or head has changed, it re-analysed here because between the last analysis and the label changing, master had obtained a few more commits.

I agree that this behaviour is undesirable (we do though want to re analyse for things like the target branch changing, even if no new code has been added). I've filed an issue internally to fix this.

**Comment by samlanning** [comment]
Ah right I see! Yes the comments can get a little too much if they're just repeated.

**Pull request merged**